File: X/python/XRPL_UKL1/soal6.py
#!/bin/python3
"""
Soal    : 6 / Soal array 2
A       : 3x2
B       : 2x3
fill    : 3x3
"""

a=[
    [1,8],
    [3,2],
    [4,6]
]
b=[
    [9,10,21], #9 + 10 = 21
    [5, 9, 8]
]
fill=[
    [0,0,0],
    [0,0,0],
    [0,0,0]
]
kolom1=len(fill)
kolom2=len(fill[0])
print("Hasil AxB")

for i in range(kolom1):
    for j in range(kolom2):
        for k in range(2):
            fill[i][j]=fill[i][j]+a[i][k]*b[k][j]


for i in range(kolom1):
    for j in range(kolom2):
        print(f"{fill[i][j]}\t", end="")
    print()

File: X/javascript/XRPL_UKL1/soal6.js
/*
Soal    : 6 / Soal array 2
A       : 3x2
B       : 2x3
fill    : 2x3
*/
A = {
    0: [1, 8],
    1: [3, 2],
    2: [4, 6]
};
B = {
    0: [9, 10, 21],
    1: [5, 9, 8],
};

fill = {
    0: [0, 0, 0],
    1: [0, 0, 0],
    2: [0, 0, 0]
};

console.log("Hasil AxB");
for (i = 0; i < 3; i++) {
    for (j = 0; j < 3; j++) {
        for (k = 0; k < 2; k++) {
            fill[i][j] = fill[i][j] + A[i][k] * B[k][j];
        }
    }
}
for (i = 0; i < 3; i++) {
    for (j = 0; j < 3; j++) {
        process.stdout.write(fill[i][j] + "\t");
    }
    console.log();
}
